examples/bidder/ad_geo.hpp

```diff
@@ -0,0 +1,91 @@
+/* 
+ * File:   ad_geo.hpp
+ * Author: arseny.bushev@gmail.com
+ *
+ * Created on 16 февраля 2017 г., 21:19
+ */
+
+#ifndef AD_GEO_HPP
+#define AD_GEO_HPP
+
+#include "config.hpp"
+/* 
+ * Geo Targeting is implemented this way to test selection and reload on big amount of records
+ */
+struct AdGeo {
+    std::string ad_id;
+    uint32_t geo_id;
+    std::string record;
+    
+    AdGeo(std::string ad_id, uint32_t geo_id) : 
+        ad_id{std::move(ad_id)}, geo_id{geo_id}, record{}
+    {}
+    AdGeo():
+        ad_id{}, geo_id{}, record{}
+    {}
+        
+    friend std::ostream &operator<<(std::ostream & os, const std::shared_ptr<AdGeo> &ad_geo_ptr)  {
+        os <<  *ad_geo_ptr;
+        return os;
+    }
+    friend std::ostream &operator<<(std::ostream & os, const  AdGeo & value)  {
+        os << value.ad_id << "|" 
+           << value.geo_id
+        ;
+        return os;
+    }
+    friend std::istream &operator>>(std::istream &is, AdGeo &l) {
+        if ( !std::getline(is, l.record) ){
+            return is;
+        }
+        std::vector<std::string> fields;
+        boost::split(fields, l.record, boost::is_any_of("\t"), boost::token_compress_on);
+        if(fields.size() < 2) {
+            std::cout << "Not enought fields " << l.record << std::endl;
+            return is;
+        }
+        l.ad_id = fields.at(0); 
+        l.geo_id = atoi(fields.at(1).c_str());
+        return is;
+    }
+};
+
+template <typename Memory = typename mpclmi::ipc::Shared, 
+          typename Alloc = typename datacache::entity_cache<Memory, ipc::data::ad_container>::char_allocator >
```

Comment:
maybe it should use ipc::data::ad_geo_container>::char_allocator ?

Even though the allocators  are the same now for all containers, maybe more readable and also what if we want to use different allocators for different containers